src/cli/commands/tasks/tasks-command.ts

```diff
@@ -39,101 +39,115 @@ export class TasksCommand {
   }
+  
+  
 
   public async handleLeagues4(options: any): Promise<ITask[]> {
-    const structId: ParamID = 873 as ParamID;
+    // console.log('handleLeagues4 invoked with options:', options);
+  
+    const structId: ParamID = PARAM_ID.LEAGUE_VARBIT_INDEX;
+    const categoryParamId: ParamID = PARAM_ID.LEAGUE_CATEGORY_ID;
+    const tierParamId: ParamID = PARAM_ID.LEAGUE_TIER_ID;
+  
+    // Define tierMap (could probably move this not here)
+    const tierMap = {
+      1: 'Easy',
+      2: 'Medium',
+      3: 'Hard',
+      4: 'Elite',
+      5: 'Master',
+      6: 'Grandmaster',
+    };
+  
+    // Fetch category enum (3413)
+    const categoryEnum = await this.enumService.getEnum(3413);
+    
+    // Ensure the categories are read and still valid from previous year.
+    console.log('Loaded categoryEnum:', Array.from(categoryEnum.map.entries()));
+
+  
+    // Sort function for tasks
     const sortFunction = (a: Struct, b: Struct) => {
       const aSort = a.params.get(structId) as number;
       const bSort = b.params.get(structId) as number;
       return aSort - bSort;
     };
-    const tierParamId: ParamID = 1852 as ParamID;
-    const easy: Struct[] = (await this.structService.findByParam(tierParamId, 1)).sort(sortFunction);
-    const medium: Struct[] = (await this.structService.findByParam(tierParamId, 2)).sort(sortFunction);
-    const hard: Struct[] = (await this.structService.findByParam(tierParamId, 3)).sort(sortFunction);
-    const elite: Struct[] = (await this.structService.findByParam(tierParamId, 4)).sort(sortFunction);
-    const master: Struct[] = (await this.structService.findByParam(tierParamId, 5)).sort(sortFunction);
+  
+    const easy = (await this.structService.findByParam(tierParamId, 1)).sort(sortFunction);
+    const medium = (await this.structService.findByParam(tierParamId, 2)).sort(sortFunction);
+    const hard = (await this.structService.findByParam(tierParamId, 3)).sort(sortFunction);
+    const elite = (await this.structService.findByParam(tierParamId, 4)).sort(sortFunction);
+    const master = (await this.structService.findByParam(tierParamId, 5)).sort(sortFunction);
+  
     const all = [...easy, ...medium, ...hard, ...elite, ...master];
     let allAsTasks: any[] = [];
-
-    if (options.legacy) {
-      allAsTasks = all.map((s, i) => {
-        const out = {
-          id: '' + (s.params.get(structId) as number),
-          monster: '',
-          name: s.params.get(874 as ParamID) as string,
-          description: s.params.get(875 as ParamID) as string,
-          category: '',
-          tier: '',
-          clientSortId: '' + i,
-        };
-        return out;
-      });
-    } else {
-      allAsTasks = all.map((s, i) => {
-        const out: ITask = {
-          structId: s.params.get(structId) as number,
-          sortId: i,
-        };
-        return out;
-      });
-    }
-
+  
+    allAsTasks = all.map((s, i) => {
+      const categoryValue = s.params.get(categoryParamId) as number;
```

Comment:
You removed the legacy vs non-legacy output, but refer to it in the readme. We can either restore it here, or we should also remove it from the readme and the leagues4 command options

Reply:
Due to a lack of full understanding around it, have reintroduced legacy for now until we can revisit it and make a call down the line.